pykafka/partition.py

```diff
@@ -140,4 +142,6 @@ def update(self, brokers, metadata):
                 log.info('Updating in sync replicas list for %s', self)
                 self._isr = [brokers[b] for b in metadata.isr]
         except KeyError:
-            raise Exception("TODO: Type this exception")
+            raise LeaderNotAvailable("Leader for partition %s not available. This is "
```

Comment:
Apart from L135, couldn't the `KeyError` also come from L143, in which case it's not a missing leader but actually another replica?